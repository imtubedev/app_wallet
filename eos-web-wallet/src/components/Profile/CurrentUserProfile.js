//@flow
import Profile from "./Profile";
import withProfile from "../../containers/profile/index";
import withUserId from "../../containers/current-user/index";

export default withUserId(withProfile(Profile));
