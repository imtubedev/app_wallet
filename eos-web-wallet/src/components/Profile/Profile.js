//@flow
import * as React from "react";
import cx from "classnames";
import css from "./styles.module.scss";
import type { UserProfile } from "../../types/UserProfile";

type Props = {
  userId: string,
  userProfile: UserProfile
};

const Profile = ({
  userId,
  userProfile: {
    eosAccount,
    imageUrl = "/images/user.png",
    location,
    displayName,
    about,
    website
  }
}: Props) => (
  <div className={cx("columns is-variable is-2 is-mobile", css.profile)}>
    <div className="column is-narrow">
      <img
        src={imageUrl === "" ? "/images/user.png" : imageUrl}
        className={css.thumbnail}
        alt={displayName || userId}
      />
    </div>
    <div className="column u-mt1">
      <div className={cx("heading is-6", css.heading)}>
        {eosAccount ? `@${eosAccount}` : userId}
      </div>
      <div className={cx("title", css.title)}>{location}</div>
      {website && (
        <a href={website} target="_blank">
          {website}
        </a>
      )}
      <p className="is-hidden-mobile">{about}</p>
    </div>
  </div>
);

export default Profile;
