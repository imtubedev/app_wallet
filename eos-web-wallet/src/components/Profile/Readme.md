Profile example:

```js
<div style={{ backgroundColor: "#3a4249", padding: "2rem"}}>
  <Profile userId="here@there.com"
    userProfile={
      {
        eosAccount: "inita",
        imageUrl: "https://bytemaster.github.io/assets/img/authorimage.jpg",
        location: "Unknown",
        about: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        website: "https://twitter.com/tandemly"
      }
    }
  />
</div>
```

Profile example (no EOS account or image):

```js
<div style={{ backgroundColor: "#3a4249", padding: "2rem"}}>
  <Profile userId="here@there.com"
    userProfile={
      {
        location: "Unknown",
        about: "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        website: "https://twitter.com/tandemly"
      }
    }
  />
</div>
```