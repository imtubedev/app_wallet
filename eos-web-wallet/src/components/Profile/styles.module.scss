@import "../../styles/custom_variables.scss";

.profile {
  .thumbnail {
    border: solid 8px $orange;
    width: 125px;
    height: 125px;
    border-radius: 75px;
  }
  .heading {
    font-size: 22px;
    font-weight: bold;
    color: $white;
    text-transform: none;
    word-break: break-all;
    letter-spacing: 0;
  }
  .title {
    font-size: 18px;
    color: $white;
    margin-bottom: 0.5rem;
  }
  a {
    word-break: break-all;
  }
}
