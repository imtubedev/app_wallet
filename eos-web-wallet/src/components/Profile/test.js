import * as React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";
import Profile from "./";

describe("<Profile />", () => {
  it("renders without crashing", () => {
    const staticContext = {};
    const profile = {
      imageUrl: "src",
      location: "Friday Harbor, WA",
      displayName: "Rick Sanchez",
      about: "active",
      website: "http://mywebsite.com/me"
    };

    const tree = renderer
      .create(
        <StaticRouter location="/" context={staticContext}>
          <Profile userProfile={profile} />
        </StaticRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
