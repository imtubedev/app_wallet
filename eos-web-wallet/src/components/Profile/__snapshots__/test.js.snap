// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders without crashing 1`] = `
<div
  className="columns is-variable is-2 is-mobile"
>
  <div
    className="column is-narrow"
  >
    <img
      alt="Rick Sanchez"
      className={undefined}
      src="src"
    />
  </div>
  <div
    className="column u-mt1"
  >
    <div
      className="heading is-6"
    />
    <div
      className="title"
    >
      Friday Harbor, WA
    </div>
    <a
      href="http://mywebsite.com/me"
      target="_blank"
    >
      http://mywebsite.com/me
    </a>
    <p
      className="is-hidden-mobile"
    >
      active
    </p>
  </div>
</div>
`;
